packages/core/src/media/metrics-new.ts

```diff
@@ -733,8 +733,73 @@ export class PlaybackEventBuffer {
   }
 
   public addEvents(events: PlaybackEvent[]) {
+    for (const newEvent of [...events]) {
+      if (newEvent.type !== "heartbeat") {
+        this.externalBuffer.push(newEvent);
+        continue;
+      }
+
+      const existingEventIndex = this.externalBuffer.findIndex(
+        (event) => event.type === "heartbeat",
+      );
+
+      if (existingEventIndex === -1) {
+        this.externalBuffer.push(newEvent);
+      } else {
+        const existingEvent = this.externalBuffer[
+          existingEventIndex
+        ] as HeartbeatEvent;
+
+        const mergedEvent: HeartbeatEvent = {
+          ...existingEvent,
+          id: existingEvent.id,
+          timestamp: Math.max(existingEvent.timestamp, newEvent.timestamp),
```

Comment:
we talked about having another timestamp field to capture the original heartbeat. I like the idea of having `timestamp` as the latest one. just need one that is set on the original event.

Reply:
This will be shipped in the next release. Thanks for the reminder